R/subsetting.R

```diff
@@ -0,0 +1,169 @@
+#' Subsetting tibbles
+#'
+#' @description
+#' Accessing columns, rows, or cells via `$`, `[[`, or `[` works very much like
+#' [with regular data frames][base::Extract.data.frame]. However, the
+#' behavior is different for tibbles and data frames in some cases:
+#' * `[` always returns a tibble by default, even if
+#'   only one column is accessed.  The `drop` argument is supported, but you
+#'   have to request it explicitly.
+#' * Partial matching of column names with `$` and `[[` is not supported, a
+#'   warning is given and `NULL` is returned.
```

Comment:
Maybe split this into two bullets: only exact matching + `$` warns if column not found

Reply:
Not sure, these two topics are very closely related, and `[[` has an `exact` argument which is ignored. But I don't have a strong opinion.